tty/pc-tty/ttypc_vtf.c

```diff
@@ -634,212 +656,340 @@ void _ttypc_vtf_sgr(ttypc_virt_t *virt)
 		case 46:
 		case 47:
 			if (ttypc->color) {
-				colortouched = 1;
-				setcolor |= ((bgansitopc[(virt->parms[i - 1] - 40) & 7]) << 8);
+				cc = 1;
+				attr = (bgansitopc[(vt->parms[i - 1] - 40) & 7] << 8) | (attr & 0x0f00);
 			}
 			break;
 		}
-	} while (i <= virt->parmi);
+	} while (i <= vt->parmi);
 	
-	if (ttypc->color) {
-		if (colortouched)
-			virt->attr = setcolor;
-		else
-			virt->attr = ((sgr_tab_color[virt->vtsgr]) << 8);
-	}
-	else
-		virt->attr = ((sgr_tab_mono[virt->vtsgr]) << 8);
+	vt->attr = (ttypc->color) ? ((cc) ? attr : csgr[vt->sgr] << 8) : msgr[vt->sgr] << 8;
 }
 
 
-/* device status reports */
-void _ttypc_vtf_dsr(ttypc_virt_t *virt)
+/* Device status reports */
+void _ttypc_vtf_dsr(ttypc_vt_t *vt)
 {
-#if 0
-	static uint8_t *answr = (uint8_t *)"\033[0n";
-	static uint8_t *panswr = (uint8_t *)"\033[?13n";      /* Printer Unattached */
-	static uint8_t *udkanswr = (uint8_t *)"\033[?21n";    /* UDK Locked */
-	static uint8_t *langanswr = (uint8_t *)"\033[?27;1n"; /* North American*/
-#endif
-#if 0
-	static uint8_t buffer[16];
-
+	static char *stat  = "\033[0n";     /* Status */
+	static char *print = "\033[?13n";   /* Printer Unattached */
+	static char *udk   = "\033[?21n";   /* UDK Locked */
+	static char *lang  = "\033[?27;1n"; /* North American */
+	char buff[16];
 	int i = 0;
 
-	switch (virt->parms[0]) {
-	
-	/* return status */
+	switch (vt->parms[0]) {
+	/* Status */
 	case 5:
-		/* respond(answr); */
+		write(0, stat, sizeof(stat));
```

Comment:
`sizeof(stat) = 4` (`sizeof(char *)`). It should be `static const char stat[]`

Reply:
Fixed.